CLR/Core/CLR_RT_Memory.cpp

```diff
@@ -16,8 +16,6 @@ static int s_PreHeapInitIndex = 0;
 
 ////////////////////////////////////////////////////////////
 
-HAL_DECLARE_CUSTOM_HEAP( CLR_RT_Memory::Allocate, CLR_RT_Memory::Release, CLR_RT_Memory::ReAllocate );

```

Comment:
Umm, hold on a sec... This is defining the managed heap APIs. The HAL_DECLARE_CUSTOM_HEAP creates a set of private_xxx APis that are inlined wrappers around the methods listed in the macro. While the entire approach to using a macro like this is dubious at best, we don't want to eliminate the functionality  as intended here. The use here is intended to be the managed heap which has lots of internal requirements and assumptions for the collector to work properly.